tests/unit/reconcile/connectors/test_databricks.py

```diff
@@ -0,0 +1,112 @@
+import re
+from unittest.mock import MagicMock, create_autospec
+
+import pytest
+from databricks.sdk import WorkspaceClient
+from pyspark.errors import PySparkException
+
+from databricks.labs.remorph.reconcile.connectors.databricks import DatabricksDataSource
+
+
+def initial_setup():
+    pyspark_sql_session = MagicMock()
+    spark = pyspark_sql_session.SparkSession.builder.getOrCreate()
+
+    # Define the source, workspace, and scope
+    source = "snowflake"
```

Comment:
Should it be different here?

Reply:
@ganeshdogiparthi-db: I think we need to change this variable name, In Datasource abstract class this purely indicates database engine and we use the engine name as prefix for getting secrets key.

As per our discussion on supporting HMS and UC within databricks source adapter we will use catalog_name and not source variable to differentiate.

Reply:
It is as expected. DatabricksDataSource will be intialized even for the source "snowflake" (because our target layer will always be databricks).  Also if you observe line 39 and 64 I have passed "databricks" as source and tested the output. (Old code)

Now up on a suggestion from Sundar. I have renamed the "source" param to the "engine". Now it has more context. I made appropriate changes to the code and test cases.